src/command/archive/push/file.c

```diff
@@ -117,21 +151,69 @@ archivePushFile(
                 compressible = false;
             }
 
-            // If there is a cipher then add the encrypt filter
-            if (cipherType != cipherTypeNone)
+            // Initialize per-repo destination files
+            StorageWrite **destination = memNew(sizeof(StorageWrite *) * repoTotal);
+
+            for (unsigned int repoIdx = 0; repoIdx < repoTotal; repoIdx++)
             {
-                ioFilterGroupAdd(
-                    ioReadFilterGroup(storageReadIo(source)),
-                    cipherBlockNew(cipherModeEncrypt, cipherType, BUFSTR(cipherPass), NULL));
-                compressible = false;
+                // Does this repo need a copy?
+                if (destinationCopy[repoIdx])
+                {
+                    // Create destination file
+                    destination[repoIdx] = storageNewWriteP(
+                        storageRepoIdxWrite(repoIdx),
+                        strNewFmt(STORAGE_REPO_ARCHIVE "/%s/%s", strZ(repoData[repoIdx].archiveId), strZ(archiveDestination)),
+                        .compressible = compressible && repoData[repoIdx].cipherType == cipherTypeNone);
+
+                    // If there is a cipher then add the encrypt filter
+                    if (repoData[repoIdx].cipherType != cipherTypeNone)
+                    {
+                        ioFilterGroupAdd(
+                            ioWriteFilterGroup(storageWriteIo(destination[repoIdx])),
+                            cipherBlockNew(
+                                cipherModeEncrypt, repoData[repoIdx].cipherType, BUFSTR(repoData[repoIdx].cipherPass), NULL));
+                    }
+                }
             }
 
-            // Copy the file
-            storageCopyP(
-                source,
-                storageNewWriteP(
-                    storageRepoWrite(), strNewFmt(STORAGE_REPO_ARCHIVE "/%s/%s", strZ(archiveId), strZ(archiveDestination)),
-                .compressible = compressible));
+            // Open source file
+            ioReadOpen(storageReadIo(source));
```

Comment:
It depends on how the source file is created, i.e. whether .ignoreMissing is specified. storageCopy() doesn't know so it has to check.

Here we do know so we don't have to check. If we added the check it would be impossible to get coverage on it.